### PR TITLE
DR-644 Truncate action history

### PR DESCRIPTION
**Pull request by timrickwood**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/actions/action-infrastructure-stack/action.yaml [hunk 1]
```diff
@@ -25,10 +25,10 @@ runs:
   steps:
       - name: "Derive TF Version"
         id: derive_tf_version
-        uses: NHSDigital/uec-dos-management/.github/actions/derive-tf-version@latest
+        uses: NHSDigital/uec-dos-management/.github/actions/derive-tf-version@v0.45
```
Comment (timrickwood) [inline .github/actions/action-infrastructure-stack/action.yaml:28]: Switch back to latest post review pre merge

File: .github/workflows/test-truncate-workflow-history.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Test Truncate Workflow History
+
+on:
+  push:
```
Comment (timrickwood) [inline .github/workflows/test-truncate-workflow-history.yaml:4]: Remove push restore schedule

File: .github/workflows/truncate-workflow-history.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    timeout-minutes: 5
+    steps:
+      - name: Delete workflow runs
+        uses: Mattraks/delete-workflow-runs@v2
+        with:
+          token: '${{ github.token }}'
+          repository: "NHSDigital/${{ inputs.domain }}"
+          retain_days: ${{ inputs.retain_days }}
+          keep_minimum_runs: ${{ inputs.minimum_history }}
+          dry_run: true
```
Comment (timrickwood) [inline .github/workflows/truncate-workflow-history.yaml:34]: Set to false post review

File: .github/workflows/truncate-workflow-history.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: Pipeline manage workflow run history
+
+on:
+  workflow_call:
+    inputs:
+      domain:
+        description: "The code repository to checkout"
+        required: false
```
Comment (timrickwood) [inline .github/workflows/truncate-workflow-history.yaml:8]: true

**Review by mmg-nhse — APPROVED** [review]

**Pull request merged**